packages/go_router/lib/src/delegate.dart

```diff
@@ -276,15 +274,14 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
           index - 1,
           context: context,
           matches: matches,
-          configuration: configuration,
         );
       }
       return SynchronousFuture<bool>(false);
     }
 
     final FutureOr<bool> exitFuture = goRoute.onExit!(
       context,
-      match.buildState(_configuration, configuration),
+      match.buildState(_configuration, currentConfiguration),
```

Comment:
so the difference is whether we use the new RouteMatchList after the change vs old RouteMatchList?

I think for redirect we use the new RouteMatchList. it seems to previous implementation is more inline with the redirect API no?

Reply:
I maybe wrong here. do you have use case that using the old routematchList is more preferred? on the other hand, I plan to redesign redirect, so maybe we should just provide both before and after?

Reply:
Yes, I found one in https://github.com/flutter/packages/pull/6614. I was making `go_router_builder` use `onExit` in some tests failed https://github.com/flutter/packages/pull/6614/checks?check_run_id=24284539541.

The issue was when using `.go()`. The `onExit` had the state of the new page, not the previous page.

The test I wrote in this PR 

```dart
'It should provide the correct path parameters to the onExit callback during a go
```

fails without my changes.

> On the other hand, I plan to redesign redirect, so maybe we should just provide both before and after?

Sure, do you want me to change something then?

Reply:
I will need to redesign the onExit in the future. It also has issue with popscope and cupertinobackswipe. Thing may become a lot different after I resolve all these issues.

After thinking about it more, I think we should just keep it simple for now. I think what you have is fine.